src/compile/repeat.ts

```diff
@@ -1,37 +1,50 @@
-import { Channel } from '../channel';
-import { CellConfig, Config } from '../config';
-import { Encoding } from '../encoding';
-import { Facet } from '../facet';
-import { Field, FieldDef, isRepeatRef } from '../fielddef';
-import { Repeat } from '../repeat';
-import { RepeatSpec } from '../spec';
-import { duplicate, vals, Dict } from '../util';
-import { VgData, VgLayout, VgScale } from '../vega.schema';
-import { buildModel } from './common';
-import { assembleData } from './data/assemble';
-import { assembleLayoutData } from './layout';
-import { Model } from './model';
+import {Channel} from '../channel';
+import {CellConfig, Config} from '../config';
+import {Encoding} from '../encoding';
+import {Facet} from '../facet';
+import {Field, FieldDef, isRepeatRef} from '../fielddef';
+import * as log from '../log';
+import {Repeat} from '../repeat';
+import {RepeatSpec} from '../spec';
+import {Dict, duplicate, vals} from '../util';
+import {VgData, VgLayout, VgScale} from '../vega.schema';
+import {buildModel} from './common';
+import {assembleData} from './data/assemble';
+import {assembleLayoutData} from './layout';
+import {Model} from './model';
 
 
 export type RepeatValues = {
-  row: string,
-  column: string
+  row?: string,
+  column?: string
 };
 
-export interface EncodingOrFacet<F> extends Encoding<F>, Facet<F> {}
+export function facetRepeatResolve(facet: Facet<Field>, repeatValues: RepeatValues): Facet<string> {
```

Comment:
facet/encodingReplaceRepeater?

I don't like resolve because we will use resolve a lot for scale / axis resolution.